graph/src/data/subgraph/schema.rs

```diff
@@ -362,3 +609,16 @@ fn set_entity_operation(
         data: data.into(),
     }
 }
+
+pub fn generate_entity_id() -> String {
+    // Fast crypto RNG from operating system
+    let mut rng = OsRng::new().unwrap();
+
+    // 128 random bits
+    let id_bytes: [u8; 16] = rng.gen();
+
+    // 32 hex chars
+    // Comparable to uuidv4, but without the hyphens,
```

Comment:
Is the likelihood (or the lack of it) comparable to uuidv4?

Reply:
Yup, likelihood of collisions is low enough to be ignored. With UUID v4, 122 bits are random, so the first duplicate ID would be expected after about 2^61 IDs have been generated. Here, we don't waste 6 bits on a version identifier, so 32 hex chars = 16 bytes = 128 bits are fully random, and the first duplicate ID is expected after about 2^64 IDs, slightly better than UUID v4.